api/config/v1alpha1/metric_types.go

```diff
@@ -44,4 +44,12 @@ type OpenTelemetrySink struct {
 }
 
 type PrometheusProvider struct {
+	// Port defines the port for Prometheus metrics.
+	// It shares the same port as EnvoyProxy readiness port by default.
+	//
+	// +optional
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=65535
+	// +kubebuilder:default=19001
+	Port int32 `json:"port,omitempty"`
```

Comment:
TBH, I don't think this's right place to put, actually prometheus reuse the port for readiness, and I think this's the right way.

Reply:
thanks, i thought we only need to make metric port configurable, and this port happen to be the same as readiness port by default.

if metric port reuse readiness port, and i will make readiness port configurable.

Reply:
I'm looking forward an API named without readiness or metric. EG maybe extends the usage in the future.

Reply:
we may need to account for both cases - metrics port and readiness port is same or different

Reply:
> we may need to account for both cases - metrics port and readiness port is same or different

I cannot image a use case for different port, it's a kind of waste.

Reply:
just thinking out loud, if we need to take both future-extensible and all-ports-are-configurable into consideration, i suppose we could make this api looks more like:
```yaml
ports:
- type: EnvoyReadiness
  name: xxx
  port: 19001  # 19001 is the default port, the user can reassign it themselves
- type: EnvoyMetrics
  name: xxx
  port: 19001 # ditto
- type: SomePortTypeWillBeUsedInTheFuture
  name: xxx
  port: xxx
```

all we need to do is to maintain the type of port.

Reply:
in this case, the xds server port (#244) can also be configured in this way (like `type: XdsServer`)

any thoughts? cc @arkodg @zirain